lib/msf/core/module_manager/cache.rb

```diff
@@ -71,32 +71,35 @@ def cache_in_memory(class_or_module, options={})
   # @param [String] reference_name the module reference name.
   # @return [false] if a module with the given type and reference name does not exist in the cache.
   # @return (see Msf::Modules::Loader::Base#load_module)
-  def load_cached_module(type, reference_name)
+  def load_cached_module(type, reference_name, use_cache: false)
     loaded = false
 
-    module_info = self.module_info_by_path.values.find { |inner_info|
-      inner_info[:type] == type and inner_info[:reference_name] == reference_name
-    }
+    if use_cache
+      cached_metadata = Msf::Modules::Metadata::Cache.instance.get_metadata_hash["#{type}_#{reference_name}"]
```

Comment:
Let's hide this implementation detail in the API

```suggestion
      cached_metadata = Msf::Modules::Metadata::Cache.instance.get_module_reference(type: type, reference_name: reference_name)
```